src/pinefarm/external/nnlojet/nnpdf_interface.py

```diff
@@ -0,0 +1,350 @@
+    if "k1" in kin_variables:
+        # Time to translate!
+        _legacy_nnpdf_translation(kin_df, metadata.process_type)
+        kin_variables = kin_df.columns.get_level_values(0)
+
+    hist_vars = list(HISTOGRAM_VARIABLES.intersection(kin_variables))
+
+    # Now preprocess the variables according to the process type
+
+    # Drell Yan
+    if process in ("WPWM", "Z0", "DY"):
+        # Special case: is a histogram binned by invariant mass?
+        if "M2" in hist_vars:
+            if len(kin_df["M2"]["mid"].unique()) == 1:
+                hist_vars.remove("M2")
+
+    # Create the histogram depending on whether this is a 1D or 2D distribution (or total)
+    histograms = None
+
+    if len(hist_vars) == 1:
+        histograms = [_1d_histogram(kin_df, hist_vars[0])]
+    elif len(hist_vars) == 2:
+        # Let's (hope) it is in M2
+        if "M2" not in hist_vars:
+            raise NotImplementedError(f"Don't know how to do this 2D: {hist_vars}")
+        hist_vars.remove("M2")
+
+        another_v = hist_vars[0]
+        # Get the unique M2 values
+        unique_m2 = kin_df["M2"]["mid"].unique()
+        m_name = _nnlojet_observable("M", process)
+        histograms = []
+        probable_bounds = np.unique(_1d_histogram(kin_df, "M2")["bins"]).tolist()
+        for i, val in enumerate(unique_m2):
+            idx = kin_df["M2"]["mid"] == val
+            tmp = _1d_histogram(kin_df[idx], another_v)
+            tmp["name"] = f"{another_v}_bin_{i}"
+            tmp["extra_selectors"] = [
+                {
+                    "observable": f"{m_name}",
+                    "min": probable_bounds[i],
+                    "max": probable_bounds[i + 1],
+                }
+            ]
+            histograms.append(tmp)
+    elif len(hist_vars) == 0 and metadata.process_type == "INC":
+        # inclusive cross section, just create a big enough histogram
+        histograms = [{"observable": "y", "bins": [-10.0, 10.0], "name": "tot"}]
+    else:
+        raise NotImplementedError(
+            "3D distributions not implemented or process not recognized"
+        )
+
+    is_normalized = metadata.theory.operation.lower() == "ratio"
+    if is_normalized:
+        print(
+            "\033[91m [WARNING] \033[0m This dataset is probably normalized, you might be missing the runcard for the fiducial cross section"
+        )
+
+    # Prepare the metadata for the data
+    hepdata = metadata._parent.hepdata.url
+    arxiv = metadata._parent.arXiv.url.split("/")[-1]
+    tables = metadata.tables
+    if not hepdata.startswith("https:"):
+        # Try to autoguess the doi
+        hepdata = f"https://doi.org/{hepdata}"
+
+    data_comment = f"arXiv number: {arxiv}, hepdata entry: {hepdata} (tables: {tables})"
+
+    # For some NNPDF datasets, different processes/energies might be grouped together
+    processes = [process]
+    if process.startswith("WPWM"):
+        processes = [process.replace("WP", ""), process.replace("WM", "")]
```

Comment:
```suggestion
```
Removing these two lines seems to solve the problem of the right ds generation for `CMS_WPWM_13TEV_ETA_WM` and `CMS_WPWM_13TEV_ETA_WP`.
Not sure, however, whether it might break something else somewhere else where this split is needed instead

Reply:
It will for all other datasets (yours are the first for which the split is done at the level of the dataset).

It needs to be modified to look at the fktables to see whether they include both (probably a `len` is enough for now though)

Reply:
Ok, maybe it does not really solve it as proc is now WPWM.

I think we might have to adopt a convention for the WPWM datasets. 

In the dataset `CMS_WPWM_13TEV_ETA` the two observables, WP and WM, are separated, hence splitting as it is done in line 319 above makes sense there. However, this is not the case for other datasets.

Reply:
To be fair, Wp ans Wm are separated everywhere. They are just concatenated.

The questions are, do we want to keep `WPWM` as the only type of dataset (since the data for Wp and Wm usually come together, I'd vote yes)

In that case, how do we distinguish.
I think having the observable as WP and WM as you did is a good way if we are consistent

(and then this if can stay BUT when the observable is WM it only generates WM)